test/build_size_test.sh

```diff
@@ -16,8 +16,12 @@ cmake_install_executorch_lib() {
   rm -rf cmake-out
 
   retry cmake -DBUCK2="$BUCK2" \
+          -DCMAKE_CXX_STANDARD=14 \
+          -DCMAKE_CXX_STANDARD_REQUIRED=ON \
+          -DCMAKE_CXX_FLAGS="-Wno-gnu -Werror -Wc++17-extensions" \
```

Comment:
what is this for ? specifically "-Wc++17-extensions" part.

can we do it without this flag?

Reply:
Updated with comments;
- Wc++17-extensions checks that we don't use c++17 features. I think it's good to have it in the test; with -Werror, this will fail compilation if we use c++17 features.
- Wno-gnu allows us to use gnu statement expressions.

Reply:
> Wno-gnu allows us to use gnu statement expressions.

What can you do with gnu statement expressions that you can't do with an immediately-invoked lambda? I haven't used them much.

Reply:
It sounds like they're quite equivalent, maybe worthwhile to change them as gnu statement expressions aren't part of standard c++